response_operations_ui/views/accounts.py

```diff
@@ -40,6 +49,58 @@ def get_my_account():
         return jsonify(e)
 
 
+@account_bp.route("/my-account", methods=["POST"])
+def update_account():
+    form = OptionsForm()
+    form_valid = form.validate()
+    if not form_valid:
+        flash("You need to choose an option")
+        return redirect(url_for("account_bp.get_my_account"))
+    else:
+        return redirect(url_for(form_redirect_mapper.get(form.data["option"])))
+
+
+@account_bp.route("/change-account-name", methods=["GET", "POST"])
+def change_account_name():
+    try:
+        form = ChangeAccountName()
+        user_id = session["user_id"]
+        user_from_uaa = uaa_controller.get_user_by_id(user_id)
+        user = {
+            "first_name": f"{user_from_uaa['name']['givenName']}",
+            "last_name": f"{user_from_uaa['name']['familyName']}",
+        }
+        payload = user_from_uaa
+        payload["name"] = {"familyName": form.data["last_name"], "givenName": form.data["first_name"]}
+        if request.method == "POST" and form.validate():
+            if (form.data["first_name"] != user["first_name"]) or (form.data["last_name"] != user["last_name"]):
+                errors = uaa_controller.update_user_account(payload)
+                if errors is None:
+                    full_name = f"{form.data['first_name']} {form.data['last_name']}"
+                    logger.info("Sending update account details email", user_id=user_id)
+                    personalisation = {
+                        "first_name": user["first_name"],
+                        "value_name": "name",
+                        "changed_value": full_name,
+                    }
+                    try:
+                        NotifyController().request_to_notify(
+                            email=user_from_uaa["emails"][0]["value"],
+                            template_name="change_account_details",
+                            personalisation=personalisation,
+                        )
+                        return logout.logout("Your name has been changed")
```

Comment:
For this, I think it'd be better to copy what the logout does then to use the logout view as a function.
it's tricky, I could be wrong about this so let me know if you think I am.  
The way I see it, while you CAN use other views as functions, I think it should be avoided so there aren't any gotchas in the future if you were to change how a view work and accidentally break another page.

If we needed that 'logout with custom flashed message' functionality for a 3rd page, I'd argue we should factor that out into it's own function and have all 3 call it.